doc/Route_priority_and_TTL.md

```diff
@@ -0,0 +1,131 @@
+# Route priority and time-to-live
+
+## Context
+In environments where the upstream connection has limited throughput (e.g. cellular or satellite), often times certain critical data needs to take precedence over the mass of normal telemetry messages to ensure speedy delivery. Prioritized messages and TTL addresses this concern by letting the operator specify which routes will be processed first, and how long messages should be retained when offline.
+
+## Glossary
+|Term       | Description   |
+|-----------|---------------|
+|*Routes*   | Declarations used to specify how messages should be passed between modules, IoT Hub, and any leaf devices.|
+|*TTL*      | Time-To-Live, used to specify how long before unsent messages expire in storage.|
+|*FSM*      | Finite state machine|
+
+
+## Design details
+Message priorities will be a new property of message routes.  This is because, when given a set of messages, both the set definition and the endpoint are necessary for the priority to be meaningful, which naturally lends to associating priorities with routes.
+This will move us towards a design where message routes not only define what messages goes where, but also when they’ll be sent.
+
+### Deployment manifest changes
+Since message priority and TTL will be associated with routes, the route section of the deployment manifest schema needs to be updated.  Routes will now be objects, each containing properties for the routing string, the priority, and the TTL.  By baking the priority into the route-spec, the customer will have the option of defining priority based on the module outputs, or by filtering for customer headers on the message.
```

Comment:
route-spec -> definition of a route?

Reply:
added an entry in glossary

---
In reply to: [420929604](https://github.com/Azure/iotedge/pull/2921#discussion_r420929604) [](ancestors = 420929604)